### PR TITLE
Changes the loading of kafka cacert from clowder

### PR DESCRIPTION
**Pull request by petrblaho**
## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [x] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by petrblaho** [comment]
https://github.com/RedHatInsights/app-common-python now has `LoadedConfig.kafka_ca()`

**Review by vkrizan — APPROVED** [review]

**Comment by petrblaho** [comment]
https://ci.ext.devshift.net/job/RedHatInsights-historical-system-profiles-backend-pr-check/291/console is CI run in `default` pool, https://ci.ext.devshift.net/job/RedHatInsights-historical-system-profiles-backend-pr-check/292/console is CI run in `real-managed-kafka` pool.... both passed.

Note that this PR changes pr_check.sh (or hsp_deploy_ephemeral_env.sh to be precise) to use `real-managed-kafka` for deployment.

File: historical_system_profiles/config.py [hunk 1]
```diff
@@ -24,6 +24,20 @@ def load_kafka_setting(env_name, default):
...
+def load_kafka_ssl(env_name, default):
+    if isClowderEnabled():
+        cfg = LoadedConfig
+
+        broker_cfg = cfg.kafka.brokers[0]
+
+        try:
+            return broker_cfg.authtype.value == "sasl"
+        except AttributeError:
+            return False
```
Comment (vkrizan) [inline historical_system_profiles/config.py:36]: This is not very safe to disable SSL on attribute error.

**Review by vkrizan — APPROVED** [review]
Other than the handling of attributes it looks okay. Ack.

File: historical_system_profiles/config.py [hunk 2]
```diff
@@ -90,7 +108,7 @@ def topic(topic):
 db_pool_timeout = int(os.getenv("HSP_DB_POOL_TIMEOUT", "5"))
 db_pool_size = int(os.getenv("HSP_DB_POOL_SIZE", "5"))
 
-bootstrap_servers = load_kakfa_setting("BOOTSTRAP_SERVERS", "kafka:29092")
+bootstrap_servers = load_kafka_setting("BOOTSTRAP_SERVERS", "kafka:29092")
 consume_topic = os.getenv("CONSUME_TOPIC", None)
```
[the remark this answers is not in the document]
Comment (petrblaho) [reply historical_system_profiles/config.py]: we can probably rename `load_kafka_setting` completely - it only gets first kafka broker's hostname and port.... 
`bootstrap_servers` is then used for kafka client so it knows where to look for kafka

File: historical_system_profiles/config.py (hunk 1 above)
Comment (petrblaho) [reply historical_system_profiles/config.py:36]: You are right... but what is the best way to solve this?

We can make the check different - check for `authtype`, if it is not None, do the read of `value` and compare it with `sasl`. This way we will not do exception driven development.

**Comment by petrblaho** [comment]
https://issues.redhat.com/browse/DRFT-1183
https://issues.redhat.com/browse/RHCLOUD-20005

**Pull request merged**